### PR TITLE
Move tests to appropriate directories

### PR DESCRIPTION
**Pull request by timfelle**
This pull request moves the tests from the root directory to the appropriate subdirectories. The tests for regression, reframe, and bench have been moved to the "tests" directory, while the unit tests have been moved to the "tests/unit" directory. This reorganization improves the structure and organization of the codebase.

Please note, I also moved the logo folder into the documentation. I dont think there were any references directly to it, but the Website might have.

### CODE REVIEW

**Comment by timofeymukha** [comment]
Very nice. I wonder if something has to be adjusted in the doc page about the unit tests. Not sure if a path is mentioned there.

**Comment by timfelle** [comment]
> Very nice. I wonder if something has to be adjusted in the doc page about the unit tests. Not sure if a path is mentioned there.

Nice catch, yes there were paths there

**Review by timofeymukha — APPROVED** [review]

**Comment by MartinKarp** [comment]
Looks good to me! Nice job Tim. Is there any CI somewhere still that run things in for example bench? I know in the DEEP-SEA project the things in bench were run in their CI in Julich.

**Comment by timfelle** [comment]
> Looks good to me! Nice job Tim. Is there any CI somewhere still that run things in for example bench? I know in the DEEP-SEA project the things in bench were run in their CI in Julich.

Well I have no idea, @njansson?

**Comment by njansson** [comment]
> > Looks good to me! Nice job Tim. Is there any CI somewhere still that run things in for example bench? I know in the DEEP-SEA project the things in bench were run in their CI in Julich.
> 
> Well I have no idea, @njansson?

It might break deep-sea, but other tests at Julich does not depend on `bench`

File: Makefile.am
Comment (njansson) [inline Makefile.am:1]: A minor comments would be if we also would like to add the regressions tests to the dist target, such that releases also includes the regression suite

**Review by njansson — APPROVED** [review]

**Review by MartinKarp — APPROVED** [review]

File: tests/reframe/checks.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class OutOfSourceAutotools(rfm.core.buildsystems.ConfigureBasedBuildSystem):
     3. Issue ``make`` to compile the code.
     '''
 
-    configuredir = variable(str, value='.')
+    configuredir = variable(str, value='../')
```
Comment (njansson) [inline tests/reframe/checks.py:33]:
```suggestion
    configuredir = variable(str, value='../../')
```

File: tests/reframe/checks.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class OutOfSourceAutotools(rfm.core.buildsystems.ConfigureBasedBuildSystem):
     3. Issue ``make`` to compile the code.
     '''
 
-    configuredir = variable(str, value='.')
+    configuredir = variable(str, value='../../')
```
Comment (njansson) [inline tests/reframe/checks.py:33]:
```suggestion
    configuredir = variable(str, value='.')
```

**Pull request merged**